src/tests/java/picard/sam/testers/SamFileTester.java

```diff
@@ -340,4 +342,8 @@ private File createInputFile() {
     public SamReader getInput() {
         return samRecordSetBuilder.getSamReader();
     }
+
+    public void setRecordsNeedSorting(final boolean recordsNeedSorting) {
```

Comment:
wouldn't it be
 `MdSamFileTester = new MdSamFileTester <>()` 
in java8?
not too bad...